js/apps/admin-ui/src/components/group/GroupPickerDialog.tsx

```diff
@@ -219,40 +223,44 @@ export const GroupPickerDialog = ({
           ))}
         </Breadcrumb>
         <DataList aria-label={t("groups")} isCompact>
-          {groups
-            .slice(groupId ? first : 0, max + (groupId ? first : 0))
-            .map((group: SelectableGroup) => (
-              <>
-                <GroupRow
-                  key={group.id}
-                  group={group}
-                  isRowDisabled={isRowDisabled}
-                  onSelect={setGroupId}
-                  type={type}
-                  isSearching={isSearching}
-                  setIsSearching={setIsSearching}
-                  selectedRows={selectedRows}
-                  setSelectedRows={setSelectedRows}
-                  canBrowse={canBrowse}
-                />
-                {isSearching &&
-                  group.subGroups?.length !== 0 &&
-                  group.subGroups!.map((g) => (
-                    <GroupRow
-                      key={g.id}
-                      group={g}
-                      isRowDisabled={isRowDisabled}
-                      onSelect={setGroupId}
-                      type={type}
-                      isSearching={isSearching}
-                      setIsSearching={setIsSearching}
-                      selectedRows={selectedRows}
-                      setSelectedRows={setSelectedRows}
-                      canBrowse={canBrowse}
-                    />
-                  ))}
-              </>
-            ))}
+          {groups.map((group: SelectableGroup) => (
+            <>
+              <GroupRow
+                key={group.id}
+                group={group}
+                isRowDisabled={isRowDisabled}
+                onSelect={(groupId) => {
+                  setFirst(0);
+                  setGroupId(groupId);
+                }}
+                type={type}
+                isSearching={isSearching}
+                setIsSearching={setIsSearching}
+                selectedRows={selectedRows}
+                setSelectedRows={setSelectedRows}
+                canBrowse={canBrowse}
+              />
+              {isSearching &&
+                group.subGroups?.length !== 0 &&
+                group.subGroups!.map((g) => (
```

Comment:
We can remove the non-null assertion operator here by improving the defensive coding we do here. Although it's more verbose, it should be more type-safe as well.
```suggestion
                group.subGroups &&
                group.subGroups.length > 0 &&
                group.subGroups.map((g) => (
```